### PR TITLE
Fix/update topic remapping

### LINKED ISSUE #289: Release/RX7 has incorrect topic mappings; JIRA Issue: #CAR-940
### Types of Issue

- [x] Anomaly report (something appears to not work correctly)
- [ ] Enhancement request (describe the enhancement being requested)
- [ ] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary
Some topics related to vehicle control are not mapped correctly in release/RX7 branch. This prevents the vehicle from moving. As the speed commands do not make it to the SSC.

### LINKED ISSUE #290: Release/RX7 Pure pursuit wrapper provides data in kph instead of mps; JIRA Issue: #CAR-941
### Types of Issue

- [x] Anomaly report (something appears to not work correctly)
- [ ] Enhancement request (describe the enhancement being requested)
- [ ] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary
Pure pursuit wrapper is outputting the incorrect units for speed. 
### Expected behavior
Speed output in mps
### Actual behavior
Speed output in kph

### LINKED ISSUE #291: Release/RX7 Random braking occurs during operation; JIRA Issue: #CAR-942
### Types of Issue

- [x] Anomaly report (something appears to not work correctly)
- [ ] Enhancement request (describe the enhancement being requested)
- [ ] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary
When the vehicle is in motion random hard braking can occur.

### PR DESCRIPTION
**Pull request by Supraluminal**
# PR Details

## Description
Fixes several issues encountered during integration testing for CARMA3 beta release: 

* Fix topic mappings and namespaces in Guidance
* Fix lack of return statement error in Autoware Plugin and CARMANodeHandle
* Fix timestamp calculation on trajectory points in AutowarePlugin
* Change Pure Pursuit Wrapper to use ns-resolution timestamps
* Change Pure Pursuit Wrapper to use speed instead of velocity
* Fix negative timestamp calculation issue in Pure Pursuit Wrapper
* Change Pure Pursuit Wrapper to work in m/s instead of kph
* Rework vehicle motion assumptions in Pure Pursuit Wrapper to match Autoware Plugin
* Correct map file loading arguments
* Remove trajectory evening function in Autoware Plugin to avoid cases where interpolation over a trajectory inflection point causes erroneous speed derivation to result in random braking
* Change trajectory executor trajectory point expiry to process nodes based on time elapsed rather than index
* Remap current speed topic correctly for usage with the Autoware Plugin UI Widget
* Add CARMA default Rviz configuration to repository

## How Has This Been Tested?

This has been tested on-vehicle, though results are preliminary as further integration testing is required.

## Types of changes
- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Related Issue
Closes #289 
Closes #290 
Closes #291 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: autoware_plugin/src/autoware_plugin.cpp [hunk 1]
```diff
@@ -66,15 +68,15 @@ namespace autoware_plugin
...
+        current_speed_ = msg->twist.linear.x;
     }
 
     std::vector<cav_msgs::TrajectoryPlanPoint> AutowarePlugin::compose_trajectory_from_waypoints(std::vector<autoware_msgs::Waypoint> waypoints)
     {
         std::vector<autoware_msgs::Waypoint> partial_waypoints = get_waypoints_in_time_boundary(waypoints, trajectory_time_length_);
         std::vector<cav_msgs::TrajectoryPlanPoint> tmp_trajectory = create_uneven_trajectory_from_waypoints(partial_waypoints);
-        std::vector<cav_msgs::TrajectoryPlanPoint> evenly_spaced_trajectory = even_trajectory(tmp_trajectory, trajectory_point_spacing_);
-        std::vector<cav_msgs::TrajectoryPlanPoint> final_trajectory = post_process_traj_points(evenly_spaced_trajectory);
+        //std::vector<cav_msgs::TrajectoryPlanPoint> evenly_spaced_trajectory = even_trajectory(tmp_trajectory, trajectory_point_spacing_);
```
Comment (msmcconnell) [inline autoware_plugin/src/autoware_plugin.cpp]: Shall we remove this function entirely now that it is unused

File: carmajava/launch/guidance.launch [hunk 2]
```diff
@@ -63,6 +63,18 @@
...
 
+  <remap from="/base/lane_waypoints_raw" to="base/lane_waypoints_raw"/>       
+  <remap from="/lane_waypoints_array" to="lane_waypoints_array"/>
+  <remap from="/traffic_waypoints_array" to="traffic_waypoints_array"/>
+  <remap from="/red_waypoints_array" to="red_waypoints_array"/>
+  <remap from="/green_waypoints_array" to="green_waypoints_array"/>
+  <remap from="trajectory" to="plan_trajectory"/>
+  <!-- TODO use arguments for top level namespace -->
+  <remap from="trajectory_plan" to="/guidance/pure_pursuit/trajectory"/>
+  <remap from="current_pose" to="/localization/ndt_pose"/>
```
Comment (msmcconnell) [inline carmajava/launch/guidance.launch]: Lets modify this to use the LOCZ namespace argument. I don't think we need to do the same for /guidance/pure_pursuit/trajectory as that is the plugin topic which will get set by plugin discovery later anyway.

File: carmajava/launch/carma_src.launch [hunk 3]
```diff
@@ -88,6 +88,10 @@
   
   <!-- Map file arealist.txt file to load -->
   <arg name="arealist_path" default="arealist.txt" doc="Path to the arealist.txt file which contains the paths and dimensions of each map cell to load"/>
+  <arg name="area" default="5x5" doc="Dimensions of the square of cells loaded at runtime using the download and arealist load types"/>
+  <arg name="load_type" default="noupdate" doc="Enum of the map loading approach to use. Can be 'download', 'noupdate', or 'arealist'"/> 
+  <arg name="single_pcd_path" default="pcd_map.pcd" doc="Path to the map pcd file if using the noupdate load type"/>
+  <arg name="area" default="1x1" doc="Dimensions of the square of cells loaded at runtime using the download and arealist load types"/>
```
Comment (qswawrq) [inline carmajava/launch/carma_src.launch]: Looks like a duplicate definition here.

Comment (msmcconnell) [reply carmajava/launch/carma_src.launch]: Duplicating the argument is required for roslaunch arg pass through unfortunately.

**Pull request merged**